Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchSubmission.java

```diff
@@ -129,18 +148,48 @@ public CredentialsProvider getCredentialsProvider() {
         return credentialsProvider;
     }
 
-    public HttpResponse getHttpResponseViaGet(String connectUrl) throws IOException {
-        CloseableHttpClient httpclient = getHttpClient();
+    public String negotiateAuthMethod(String connectUrl) throws IOException {
+        List<Header> replaceHeader = new ArrayList<>();
+        replaceHeader.add(new BasicHeader("User-Agent", "Mozilla/5"));
```

Comment:
The User agent header can be separated into a standalone class `BrowserUaHeader`